plugin-dev/Source/Sentry/Private/Desktop/SentrySubsystemDesktop.cpp

```diff
@@ -73,7 +80,26 @@ sentry_value_t HandleBeforeCrash(const sentry_ucontext_t *uctx, sentry_value_t e
 
 	FSentryCrashContext::Get()->Apply(SentrySubsystem->GetCurrentScope());
 
-	return HandleBeforeSend(event, nullptr, closure);
+	TSharedPtr<SentryEventDesktop> eventDesktop = MakeShareable(new SentryEventDesktop(event, true));
+
+	SentrySubsystem->GetCurrentScope()->Apply(eventDesktop);
+
+	if(!IsGarbageCollecting())
+	{
+		USentryEvent* EventToProcess = NewObject<USentryEvent>();
+		EventToProcess->InitWithNativeImpl(eventDesktop);
+
+		return SentrySubsystem->GetBeforeSendHandler()->HandleBeforeSend(EventToProcess, nullptr)
+			? event
+			: sentry_value_new_null();
+	}
+	else
+	{
+		// If crash occured during garbage collection we can't just obtain a GC lock like with normal events
+		// since there is no guarantee it will be ever freed. In this case crash event will be reported
+		// without calling a `beforeSend` handler.
```

Comment:
That's a pretty huge gotcha that we should put on the docs.

Reply:
How about adding this to the "Known limitations" section in readme instead?

Reply:
Sounds good to me! Thanks!